### ISSUE TITLE
Check common kerns are present

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#2739]
https://github.com/andre-fuchs/kerning-pairs can be used to check fonts have the most common kerns, and warn if not

**Comment by bobh0303** [comment]
for Latin script.

**Comment by davelab6** [comment]
@vv-monsalve please can you share a list/spreadsheet with the absolutely most important kern pairs?

**Comment by felipesanches** [comment]
This font family may be useful for testing the proposed new check: 
https://github.com/google/fonts/issues/1690

**Comment by davelab6** [comment]
https://twitter.com/Tosche_E/status/1282309946462306304?s=19

There are a lot of kerning pairs which seem illogical at first, and some prescriptivist designers will say "should not be kerned", but get used in the real world with disappointing results

**Comment by bobh0303** [comment]
And can you please skip this test for fonts are targeted at non-latin scripts but which include Latin-1 as a user convenience (so things like system menus work).

**Comment by m4rc1e** [comment]
Perhaps throwing a font at a list of kerning pairs is too simplistic? I've found serif and sans will often require different pairs. Many false positives will also be raised for display families.

If we do implement this check, can it raise a WARN instead of a FAIL?

**Comment by felipesanches** [comment]
yeah, I think we can start at WARN level and only consider bumping it up to FAIL level after we feel confident about the quality of the results on font files in the wild

**Comment by felipesanches** [comment]
> https://twitter.com/Tosche_E/status/1282309946462306304?s=19

Toshi's tweet is very funny, but my wife last dinner said I'm way too geek to laugh at this kind of stuff :-P

**Comment by vv-monsalve** [comment]
> @vv-monsalve please can you share a list/spreadsheet with the absolutely most important kern pairs?

Sure, I'll create one while working on Big Shoulders (haven't saw your message before).

> If we do implement this check, can it raise a WARN instead of a FAIL?

I agree some of them could depend on many factors. Nevertheless, there would be universal cases like Vo Wo To AV, not so dependant on discretional factors,  that is really painful to found in any scenario. 

Maybe the check could inspect if there is a kerning value for the desired list we built, and the output rationale for the Warn could suggest paying attention to them.

At the same time, I've been wondering what about cases where there is a value but it's exaggerated or _shouldn't be needed_ for a certain par.

![Screen Shot 2020-07-13 at 12 39 52](https://user-images.githubusercontent.com/48698976/87339297-0442fd80-c50c-11ea-99b4-fe38be7294d4.png)

**Comment by davelab6** [comment]
https://twitter.com/typegirl/status/1294431710931587075?s=19

> Here's a kern pair that needs adding to your lists. The period [.] followed by any superscripted figures.

**Comment by vv-monsalve** [comment]
> I agree some of them could depend on many factors. Nevertheless, there would be universal cases like Vo Wo To AV, not so dependant on discretional factors, that is really painful to found in any scenario.

I've created and debugged four first lists with the minimum (the musts) kerning pairs that should be covered for a Latin font.

The lists only display the actual pair of letter-letter or letter-punctuation mark and no sample texts given they were thought as sources for a new Fontbakery check.

The lists were pushed to my fork of gf-docs. Despite having checked them a couple of times to leave there the ones the more universal that are not related to specific cases (design particularities), I consider the files a WIP, as new pairs needed could be found. Suggestions are more than welcome.

- [GF_UC_to_UC.txt](https://github.com/vv-monsalve/gf-docs/blob/kerning-pairs/KerningPairs/GF_UC_to_UC.txt)
- [GF_UC_to_lc.txt](https://github.com/vv-monsalve/gf-docs/blob/kerning-pairs/KerningPairs/GF_UC_to_lc.txt)
- [GF_lc_to_lc.txt](https://github.com/vv-monsalve/gf-docs/blob/kerning-pairs/KerningPairs/GF_lc_to_lc.txt)
- [GF_lc-UC_punct.txt](https://github.com/vv-monsalve/gf-docs/blob/kerning-pairs/KerningPairs/GF_lc-UC_punct.txt)

**Comment by yanone** [comment]
This list is indeed very extensive. 

+1 for raising WARNs, but I do feel that FAILs are needed for some very important pairs, especially uppercase `TAT`, `YAY`, `WAW`, `VAV`, `LT` for example. These should work for almost every design. 

Very early in my work for GF I pushed some unimportant fonts with these pairs missing and I felt very shitty about it but was overwhelmed by the work at the time. A straight FAIL in my face would have helped me.

**Comment by RosaWagner** [comment]
I find the lists useful in the gfdocs as helpers to designers. Although I wouldn't make a fontbakery check out of it for several reasons. FB is QA tool for production: while we can find way to measure the quality of production, it is difficult to evaluate good design, and kerning is about good design. I am afraid if we have a warning with an extensive list of kerning pairs, it will end up ignored everytime like the "missing kerning for non-ligated sequences"… 

- Maybe we could try to find extreme mistakes, like "the space between those 2 glyphs is very wide compared to the average spacing value", or maybe "These pair is clashing in very unusual way compare to the rest of the font". but I think we already have warn for this, right ?
- We could encourage designers with a WARN to not kern between scripts (to have different kerning groups for A and Alpha for example), in order to save some weight.

We talked about it with Vivi and arrived to this conclusion:
- A very small list of often-kerned capital pairs as Yanone suggested could be nice for a FAIL. Clearly, if these pairs are not kerned, it means there is a problem (except for monospace fonts of course…). This is my personal checking strings for latin, Greek and Cyrillic. I used them just to see if kerning has compiled and if it is displaying correctly in Word : `HHVAHHLYHHTJHHUAHHTATHH` `ΗΗΑΥΗΗΤΛΗΗΓΔΗΗ` `ННГАННРЛННАТАННГДНН`.
- GF script for checking kerning pairs is quite overwhelming, we could update the script using Viviana's lists to give to the designer as a helper in the design process. 
- We can still put the lists in Gf-docs, it will be a nice helper for people not using GF scripts. We can elaborate it a bit too: give them links to tools helping you for kerning + encouragement to make proper spacing in the first place (this is something that I wrote for my students, it can be used in gfdocs too:  [Spacing.md.zip](https://github.com/googlefonts/fontbakery/files/5607077/Spacing.md.zip))

**Comment by davelab6** [comment]
I'm happy with that proposal @RosaWagner 

I agree with @bobh0303 - I propose the check logic should downgrade from FAIIL to WARN if a non-latin glyph set is included and the number of Latin glyphs is under 250
